caravel/assets/javascripts/explorev2/stores/store.js

```diff
@@ -11,48 +11,137 @@ export const fieldTypes = [
   'TextField',
 ];
 
-// TODO: add datasource_type here after druid support is added
 export const defaultFormData = {
-  sliceId: null,
-  vizType: null,
-  timeColumn: null,
-  timeGrain: null,
-  groupByColumns: [],
+  slice_name: null,
+  slice_id: null,
+  viz_type: 'table',
   metrics: [],
-  since: null,
-  until: null,
-  having: null,
-  where: null,
+  order_by_cols: [],
+  metric: null,
+  stacked_style: 'stack',
+  linear_color_scheme: 'blue_white_yellow',
+  normalize_across: 'heatmap',
+  horizon_color_scale: 'series',
+  canvas_image_rendering: 'pixelated',
+  xscale_interval: '1',
+  yscale_interval: '1',
+  bar_stacked: false,
+  show_markers: false,
+  show_bar_value: false,
+  order_bars: false,
+  show_controls: false,
+  reduce_x_ticks: false,
+  include_series: false,
+  secondary_metric: null,
+  country_fieldtype: 'cca2',
+  groupby: [],
   columns: [],
-  orderings: [],
-  timeStampFormat: 'smart_date',
-  rowLimit: 50000,
-  searchBox: false,
-  whereClause: '',
-  havingClause: '',
-  filters: [],
+  all_columns: [],
+  all_columns_x: null,
+  all_columns_y: null,
+  druid_time_origin: null,
+  bottom_margin: 'auto',
+  granularity: '1 day',
+  domain_granularity: 'month',
+  subdomain_granularity: 'day',
+  link_length: '200',
+  charge: '-500',
+  granularity_sqla: null,
+  resample_rule: null,
+  resample_how: null,
+  resample_fillmethod: null,
+  since: '7 days ago',
+  until: 'now',
+  max_bubble_size: '25',
+  whisker_options: 'Tukey',
+  treemap_ratio: 0.5 * (1 + Math.sqrt(5)),
+  number_format: '.3s',
+  row_limit: 50000,
+  limit: 50,
+  timeseries_limit_metric: null,
+  rolling_type: 'None',
+  rolling_periods: null,
+  series: null,
+  entity: null,
+  x: null,
+  y: null,
+  size: null,
+  url: 'https: //www.youtube.com/embed/JkI5rg_VcQ4',
+  x_axis_label: null,
+  y_axis_label: null,
+  where: '',
+  having: '',
+  compare_lag: null,
+  compare_suffix: null,
+  table_timestamp_format: 'smart_date',
+  series_height: 25,
+  x_axis_format: 'smart_date',
+  y_axis_format: '.3s',
+  markup_type: 'markdown',
+  rotation: 'random',
```

Comment:
most of the defaults should be populated already. so you might be able to get rid of this altogether.